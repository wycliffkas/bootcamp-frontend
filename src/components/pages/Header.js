import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logOut } from '../../actions/questions';

const Header = (props) => {

	let token = localStorage.getItem("token");
	const onClick = () =>{
		props.logOut();
	};

	// if(!token) {
	// 	return (
	// 		<nav className="navbar navbar-inverse">
	// 		<div className="container-fluid">
	// 		<div className="navbar-header">
	// 			<NavLink className="navbar-brand" to="/">Stack Overflow Lite</NavLink>
	// 		</div>   

			
	
	// 		</div>
	// 		</nav>
	// 	);		
	// }
	// else {
		return (
			<nav className="navbar navbar-inverse">
			<div className="container-fluid">
			<div className="navbar-header">
				<NavLink className="navbar-brand" to="/">Stack Overflow Lite</NavLink>
			</div>   
		{
			!token
			?(<ul className="nav navbar-nav navbar-right">
			<li><NavLink to="/register"> Register</NavLink></li>
			<li><NavLink to="/login"> Login</NavLink></li>   			
		</ul>)
			: (
				<div>	<ul className="nav navbar-nav navbar-left">
			<li><NavLink to="/addQuestion">Add Question</NavLink></li>
		</ul>

		<ul className="nav navbar-nav navbar-right">
			<li><a href="#" id="logOut" onClick={() => onClick()}>Logout</a></li>   				
		</ul>
		</div>) 
		}
			</div>
			</nav>
		);			
	// }

};
export { Header }

export default connect(null, { logOut })(Header);