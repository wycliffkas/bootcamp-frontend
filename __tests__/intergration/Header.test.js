import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import  { Header }  from "../../src/components/pages/Header";
import   Headerrr   from "../../src/components/pages/Header";

describe("Header", () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialUserState = {
    questions:[],
    question:{},
    loading:true,
    response:{}   
  };

  const store = mockStore({ ...initialUserState });
  const editor = shallow(<Header store={store} logOut={jest.fn()} />);
 
  it("should render without crashing", () => {
    expect(editor).toMatchSnapshot();
  });


  // it("should render a button that handles click actions", () => {
  //   const editorr = shallow(<Header store={store} logOut={jest.fn()} />);
  //   expect(
  //     editorr.find('#logOut').length).toEqual(1)
  // });
});