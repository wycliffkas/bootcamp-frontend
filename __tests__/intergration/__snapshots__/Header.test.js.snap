// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logOut={[MockFunction]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid"
      >
        <div
          className="navbar-header"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="navbar-brand"
            to="/"
          >
            Stack Overflow Lite
          </NavLink>
        </div>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/register"
            >
               Register
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/login"
            >
               Login
            </NavLink>
          </li>
        </ul>
      </div>,
      "className": "navbar navbar-inverse",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="navbar-header"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="navbar-brand"
              to="/"
            >
              Stack Overflow Lite
            </NavLink>
          </div>,
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/register"
              >
                 Register
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/login"
              >
                 Login
              </NavLink>
            </li>
          </ul>,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="navbar-brand"
              to="/"
            >
              Stack Overflow Lite
            </NavLink>,
            "className": "navbar-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Stack Overflow Lite",
              "className": "navbar-brand",
              "to": "/",
            },
            "ref": null,
            "rendered": "Stack Overflow Lite",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/register"
                >
                   Register
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/login"
                >
                   Login
                </NavLink>
              </li>,
            ],
            "className": "nav navbar-nav navbar-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/register"
                >
                   Register
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": " Register",
                  "to": "/register",
                },
                "ref": null,
                "rendered": " Register",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/login"
                >
                   Login
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": " Login",
                  "to": "/login",
                },
                "ref": null,
                "rendered": " Login",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid"
        >
          <div
            className="navbar-header"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="navbar-brand"
              to="/"
            >
              Stack Overflow Lite
            </NavLink>
          </div>
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/register"
              >
                 Register
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/login"
              >
                 Login
              </NavLink>
            </li>
          </ul>
        </div>,
        "className": "navbar navbar-inverse",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="navbar-header"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="navbar-brand"
                to="/"
              >
                Stack Overflow Lite
              </NavLink>
            </div>,
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/register"
                >
                   Register
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/login"
                >
                   Login
                </NavLink>
              </li>
            </ul>,
          ],
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="navbar-brand"
                to="/"
              >
                Stack Overflow Lite
              </NavLink>,
              "className": "navbar-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Stack Overflow Lite",
                "className": "navbar-brand",
                "to": "/",
              },
              "ref": null,
              "rendered": "Stack Overflow Lite",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/register"
                  >
                     Register
                  </NavLink>
                </li>,
                <li>
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/login"
                  >
                     Login
                  </NavLink>
                </li>,
              ],
              "className": "nav navbar-nav navbar-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/register"
                  >
                     Register
                  </NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": " Register",
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": " Register",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/login"
                  >
                     Login
                  </NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": " Login",
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": " Login",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
